src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbConnectionInternal.cs

```diff
@@ -268,25 +273,29 @@ internal bool IsEmancipated
                 //       (DbConnectionPool.Clear and ReclaimEmancipatedObjects
                 //       do this for us)
 
-                // Remember how this works (I keep getting confused...)
+                // The functionality is as follows:
                 //
                 //    _pooledCount is incremented when the connection is pushed into the pool
                 //    _pooledCount is decremented when the connection is popped from the pool
                 //    _pooledCount is set to -1 when the connection is not pooled (just in case...)
                 //
                 // That means that:
                 //
-                //    _pooledCount > 1    connection is in the pool multiple times (this is a serious bug...)
+                //    _pooledCount > 1    connection is in the pool multiple times (This should not happen)
                 //    _pooledCount == 1   connection is in the pool
                 //    _pooledCount == 0   connection is out of the pool
                 //    _pooledCount == -1  connection is not a pooled connection; we shouldn't be here for non-pooled connections.
-                //    _pooledCount < -1   connection out of the pool multiple times (not sure how this could happen...)
+                //    _pooledCount < -1   connection out of the pool multiple times
                 //
                 // Now, our job is to return TRUE when the connection is out
                 // of the pool and it's owning object is no longer around to
                 // return it.
 
+                #if NETFRAMEWORK
                 return !IsTxRootWaitingForTxEnd && (_pooledCount < 1) && !_owningObject.TryGetTarget(out _);
+                #else
+                return (_pooledCount < 1) && !_owningObject.TryGetTarget(out _);
+                #endif
```

Comment:
This behavioural difference pre-dates the initial commit of SqlClient, so it's not just a matter of a PR making a change to netfx but not to netcore.

I think this is actually a bugfix in netfx which wasn't cross-ported to netcore. `IsTxRootWaitingForEnd` just returns whether or not this is currently in stasis, and the "is in stasis" flag is set by `SetInStasis`. That method is called within DbConnectionPool, and has this comment:
> // SQLHotfix# 50003503 - connections that are affiliated with a 
> //   root transaction and that also happen to be in a connection 
> //   pool that is being shutdown need to be put in stasis so that 
> //   the root transaction isn't effectively orphaned with no 
> //   means to promote itself to a full delegated transaction or
> //   Commit or Rollback

Would you mind digging into that hotfix to see whether or not this is correct? If so, we should make .NET Core behave in the same way - whether as part of this PR, or as a separate one-line PR.

Reply:
So, I think the older ticket systems (buvsts, sqlhotfix, etc) don't exist anymore, and as such I don't think I can pull up anymore information on this. I'm checking with the team to see if there are archived copies of the tickets somewhere, but I suspect we're on our own now.

Personally I think it makes sense that if the there is a tx that the connection is waiting on, then the connection isn't emancipated. But I genuinely don't understand the intricacies of what it means to be emancipated, in stasis, etc...

Reply:
The .NET Framework behaviour definitely makes sense, and it's very old behaviour. The oldest version of the reference source is .NET 4.5.1, and the netfx logic's present in that version; it also exists in the .NET 3.5 SP1 assemblies.

I've spent more time investigating, and I think there's another possibility. System.Transactions wasn't implemented until .NET Core 2.0, and System.Data.SqlClient predates that. Maybe System.Data.SqlClient was ported to .NET Core without this logic, (because transactions weren't supported by the underlying runtime) and when support for transactions was added it just wasn't added back to DbConnectionInternal. If that's the case, we should be fine to use the .NET Framework behaviour.

Reply:
> I've spent more time investigating, and I think there's another possibility. System.Transactions wasn't implemented until .NET Core 2.0, and System.Data.SqlClient predates that. Maybe System.Data.SqlClient was ported to .NET Core without this logic, (because transactions weren't supported by the underlying runtime) and when support for transactions was added it just wasn't added back to DbConnectionInternal. If that's the case, we should be fine to use the .NET Framework behaviour.

That is the case. Distributed transactions came in .NET 7 so the original SDS Core code was likely missing some bits.

Reply:
Ok, I adopted the netfx implementation of the return statement. Let me know if there's any issues with that.

Reply:
Thanks - that explanation makes sense, and I agree that netfx has the better implementation to use.